sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -93,7 +93,14 @@ private[sql] class DataFrameImpl protected[sql](
       queryExecution.analyzed.resolve(n.name, sqlContext.analyzer.resolver).get
     }
   }
-
+ 
+  protected[sql] def numericColumns(colNames: String*): Seq[Expression] = {
```

Comment:
i don't think you'd want this. if an user explicitly specifies a column, we should just throw an error rather than silently dropping the column from numeric aggregation